optimism/src/main.rs

```diff
@@ -124,7 +136,7 @@ pub fn main() -> ExitCode {
             next: std::array::from_fn(|_| Vec::with_capacity(domain_size)),
         };
 
-    while !env.halt {
+    while !term.load(Ordering::Relaxed) && !env.halt {
```

Comment:
I'm not exactly sure I understand what's happening... I assume after registering the hook, SIGINT will not work as usual, but will instead just modify the mutable variable? 
- In this case: Is it intended that the Keccak proof generation (lines 256-270) do not depend on `term`? Will that part be interrupted by SIGINT?